### PR TITLE
Track input processedBytes with MSQ ingestion

### PR DESCRIPTION
**Pull request by kfaraz**
Follow up to #13520 

### Description
Bytes processed are currently tracked for intermediate stages in MSQ ingestion. This PR adds the capability to track the bytes processed by an MSQ controller task while reading from an external input source.

### Changes
- Create an instance of `InputStats` for every `InputSource` in `ExternalInputSliceReader`
- Track `processedBytes` in `InputStats` by invoking `InputSourceReader.read(inputStats)`
- Update `ChannelCounters` with the above obtained `processedBytes` when incrementing the input file count.
- Add expected counters in `MSQTester` to be verified in tests
- Update task report structure in docs

### Total input processed bytes
The total input processed bytes can be obtained by summing the `processedBytes` as follows:
```
totalBytes = 0
for every root stage (i.e. a stage which does not have another stage as an input):
    for every worker in that stage:
        for every input channel: (i.e. channels with prefix "input", e.g. "input0", "input1", etc.)
            totalBytes += processedBytes
```

### Release note

Track bytes read from an input source in the stage counters of an MSQ task report.

### Screenshots
`processedBytes` with sample "wikipedia" datasource:

- MSQ batch:
msq_bytes

- Classic batch:
batch_bytes

<hr>

This PR has:

- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [x] added documentation for new or modified features or behaviors.
- [x] a release note entry in the PR description.
- [x] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

File: extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/MSQTestBase.java [hunk 1]
```diff
@@ -613,6 +617,11 @@ public IngestTester testIngestQuery()
     return new IngestTester();
   }
 
+  public static CounterSnapshotBuilder channelCountersWith()
```
Comment (cryptoe) [inline extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/MSQTestBase.java:620]: Nit: syle I think CounterSnapshotBuilder can be a new class of its own and then this method gets pushed there.

**Review by cryptoe — APPROVED** [review]
LGTM thanks @kfaraz for this useful metric.

File: extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/MSQTestBase.java (hunk 1 above)
Comment (kfaraz) [reply extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/MSQTestBase.java:620]: Sure, I will move it out.

**Comment by kfaraz** [comment]
Added some more validations for SELECT and REPLACE.
The bytes are currently being validated in tests only for external input, not for segment input. 

To validate bytes read from the segment, the following things might need to be done:
- `QueryableIndex` could expose a `getSize()`
- the segments created during the initial setup of `SqlTestFramework.createQuerySegmentWalker()` could be reused in the `MSQTestBase.getSupplierForSegment()`
- the corresponding `DataSegment` objects should be retained in the `BaseCalciteQueryTest` to be used for expected size.
Since doing all of this would be outside the scope of this PR, we can address it separately.

**Pull request merged**